### PR TITLE
Add script to print history of a text doc

### PR DESCRIPTION
**Pull request by Nateowami**
Likely more scripts to follow. Hence the use of a utils file and an attempt to centralize configurations regarding where the connections should be made to. The idea is that it should be very easy to run the scripts against localhost and then switch to running them against qa or live with only minimal changes (experience has shown that constantly having to edit the scripts to specify a different port number is slow and confusing).

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1109)

### CODE REVIEW

**Review by irahopkinson — CHANGES_REQUESTED** [review]
Could you make sure the info that was in the other files gets into a GDoc along with info about how to run this and share that with the in-house devs.

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-:-MgJwmQpAhEzeqZlj5oS:b164eoz)* status: all files reviewed, 2 unresolved discussions (waiting on @Nateowami)

___
*[scripts/db_tools/text-doc-history.js, line 38 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgJw9bpEdOgrpbnuWKK:-MgJw9bpEdOgrpbnuWKL:b-e16cl9) ([raw file](https://github.com/sillsdev/web-xforge/blob/e268ca007b37e99c652c2eb8ca2f1ee591d2ec18/scripts/db_tools/text-doc-history.js#L38)):*
> ```
> lastLoggedDoc
> ```
This doesn't seem to be named right since it gets updated to the previous index on every loop rather than just when a log is made.
___
*[scripts/db_tools/text-doc-history.js, line 47 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgJvM4Z2dIXYLIc5lSS:-MgJvM4_4lIh0TOpMWcd:b-9j4xoa) ([raw file](https://github.com/sillsdev/web-xforge/blob/e268ca007b37e99c652c2eb8ca2f1ee591d2ec18/scripts/db_tools/text-doc-history.js#L47)):*
> ```
>       } else {
>         if (docs[i + 1] && docs[i + 1].m.uId === doc.m.uId) {
> ```
nit: seems unnecessary to add another level off indent when you could use `} else if (...) {`.

**Review by Nateowami — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1109)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @irahopkinson)

___
*[scripts/db_tools/text-doc-history.js, line 38 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgJw9bpEdOgrpbnuWKK:-MgNXb_U9dSUa3CprPZ2:b-2aaotc) ([raw file](https://github.com/sillsdev/web-xforge/blob/e268ca007b37e99c652c2eb8ca2f1ee591d2ec18/scripts/db_tools/text-doc-history.js#L38)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> lastLoggedDoc
> ```
This doesn't seem to be named right since it gets updated to the previous index on every loop rather than just when a log is made.
</blockquote></details>

Actually it only gets updated when logging a doc. There's a `continue` statement in the `else if`.
___
*[scripts/db_tools/text-doc-history.js, line 47 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgJvM4Z2dIXYLIc5lSS:-MgNXO_M6oAKYLPcJetP:b-l2tmhf) ([raw file](https://github.com/sillsdev/web-xforge/blob/e268ca007b37e99c652c2eb8ca2f1ee591d2ec18/scripts/db_tools/text-doc-history.js#L47)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
>       } else {
>         if (docs[i + 1] && docs[i + 1].m.uId === doc.m.uId) {
> ```
nit: seems unnecessary to add another level off indent when you could use `} else if (...) {`.
</blockquote></details>

Good point. Done. (When I wrote it this tree of logic is what made sense to me, but this is identical and makes sense too).
___
*[scripts/db_tools/utils.js, line 51 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgNXCgQ2YyWb0KLSfER:-MgNXCgQ2YyWb0KLSfES:bakj7fg) ([raw file](https://github.com/sillsdev/web-xforge/blob/e268ca007b37e99c652c2eb8ca2f1ee591d2ec18/scripts/db_tools/utils.js#L51)):*
> ```JavaScript
> module.exports = {
>   requireFromRealTimeServer,
>   fetchDoc,
> ```

I missed this. I had defined `fetchDoc`, then removed it because it wasn't used by this script. Unfortunately I left it here by mistake.

**Review by irahopkinson — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1109)* status: all files reviewed, 1 unresolved discussion (waiting on @Nateowami)

___
*[scripts/db_tools/README.md, line 6 at r3](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgOTD3Y72IgBOX6Rfre:-MgOTD3Y72IgBOX6Rfrf:b99ncqc) ([raw file](https://github.com/sillsdev/web-xforge/blob/7d78650a12023001092605b2887388215965f018/scripts/db_tools/README.md#L6)):*
> ```
> Run the script.
> ```
How do you run the script?
___
*[scripts/db_tools/text-doc-history.js, line 38 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgJw9bpEdOgrpbnuWKK:-MgOTJd17A8PFYgrayWG:bvsz65o) ([raw file](https://github.com/sillsdev/web-xforge/blob/e268ca007b37e99c652c2eb8ca2f1ee591d2ec18/scripts/db_tools/text-doc-history.js#L38)):*
<details><summary><i>Previously, Nateowami (Nathaniel Paulus) wrote…</i></summary><blockquote>

Actually it only gets updated when logging a doc. There's a `continue` statement in the `else if`.
</blockquote></details>

Right  you are (strike 2 for me).

**Review by Nateowami — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1109)* status: all files reviewed, 1 unresolved discussion (waiting on @irahopkinson)

___
*[scripts/db_tools/README.md, line 6 at r3](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgOTD3Y72IgBOX6Rfre:-MgRJqesCVCKvARZ1iB7:b-hsgwid) ([raw file](https://github.com/sillsdev/web-xforge/blob/7d78650a12023001092605b2887388215965f018/scripts/db_tools/README.md#L6)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> Run the script.
> ```
How do you run the script?
</blockquote></details>

`./script_name.js`

Actually it might only run on *nix environments. How do you execute a JS file on Windows? Does Windows respect shebangs (`#! /blah`)?

**Review by irahopkinson — APPROVED** [review]
:lgtm: lets get this merged as is.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-:-MgeREXb5tC8gh8wa4Pz:bknw48b)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @Nateowami)

___
*[scripts/db_tools/README.md, line 6 at r3](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgOTD3Y72IgBOX6Rfre:-MgePb7x4mDt7zryiUk-:b-yyzxzs) ([raw file](https://github.com/sillsdev/web-xforge/blob/b89aa088bf7a9a594dd626bf32c1ec5d5c79c6b8/scripts/db_tools/README.md#L6)):*
<details><summary><i>Previously, Nateowami (Nathaniel Paulus) wrote…</i></summary><blockquote>

`./script_name.js`

Actually it might only run on *nix environments. How do you execute a JS file on Windows? Does Windows respect shebangs (`#! /blah`)?
</blockquote></details>

In Windows I do `node ./script_name.js`. If I just call the script it defaults to opening it in VSCode. I don't know about shebangs in Windows (or Linux).

**Review by Nateowami — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1109)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[scripts/db_tools/README.md, line 6 at r3](https://reviewable.io/reviews/sillsdev/web-xforge/1109#-MgOTD3Y72IgBOX6Rfre:-Mgf2O8M4xI55e9N7DzQ:bvqbmun) ([raw file](https://github.com/sillsdev/web-xforge/blob/b89aa088bf7a9a594dd626bf32c1ec5d5c79c6b8/scripts/db_tools/README.md#L6)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

In Windows I do `node ./script_name.js`. If I just call the script it defaults to opening it in VSCode. I don't know about shebangs in Windows (or Linux).
</blockquote></details>

I just looked into this a bit and it looks like Windows ignores shebangs. It's a *nix thing, but not actually part of the POSIX standard.

Basically if a file is marked as executable and the first line starts with `#!`, then the rest of the line is considered the command that should run the file.

On Windows, running `node script_name.js` will cause Node to run it, and Node will ignore the shebang.
